agent/agent.go

```diff
@@ -1254,6 +1260,45 @@ func (a *Agent) clearGitConfig(tc *taskContext) {
 		return
 	}
 	logger.Info("Cleared git credentials.")
+
+	if err := unregisterScalar(); err != nil {
+		logger.Error(errors.Wrap(err, "unregistering scalar repositories"))
+		return
+	}
+}
+
+// unregisterScalar unregisters a repository that was cloned (and therefore registered with) Scalar.
```

Comment:
small nit: move the `with` outside the parentheses (but ty for the really understandable/context in the comment! and adding it to that cleargitconfig func too)